runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5278,12 +5285,21 @@ class INTERPRETER_CLASS
 		recordJNIReturn(REGISTER_ARGS, bp);
 		restoreSpecialStackFrameLeavingArgs(REGISTER_ARGS, bp);
 
-#if JAVA_SPEC_VERSION >= 20
+#if JAVA_SPEC_VERSION >= 21
 		/* Set the execution state after the downcall as required in the linker options. */
 		if (NULL != returnState) {
-			*returnState = (I_32)errno;
+		/* The error code layout on Windows in JDK21 is changed to a segment for an array of integers
+		 * which saves the return value of GetLastError(), WSAGetLastError() and errno.
+		 */
+#if defined(J9VM_ENV_DATA64) && defined(WIN32)
```

Comment:
Why is `defined(J9VM_ENV_DATA64)` considered here?

Reply:
`J9VM_ENV_DATA64`  is checked given the FFI specific code only works on Windows 64bit.

Reply:
While I understand we don't support Java 21+ on 32-bit Windows today (and we may never), it's inconsistent with `CapturableState` which just checks `IS_WINDOWS`; `defined(J9VM_ENV_DATA64)` should be removed from the conditions on lines 28, 34, 5294, 5298 and 5300.